src/rust/engine/src/context.rs

```diff
@@ -249,7 +264,38 @@ impl Core {
     }
     let graph = Arc::new(InvalidatableGraph(Graph::new()));
 
-    let http_client = reqwest::Client::new();
+    // These certs are for downloads, not to be confused with the ones used for remoting.
+    let ca_certs = if let Some(ref path) = ca_certs_path {
+      let mut buf = Vec::new();
+      std::fs::File::open(path)
+        .and_then(|mut f| f.read_to_end(&mut buf))
+        .map_err(|err| format!("Error reading root CA certs file {:?}: {}", path, err))?;
+      let content = std::str::from_utf8(&buf)
+        .map_err(|err| format!("Error decoding root CA certs file {:?}: {}", path, err))?;
```

Comment:
I think that this can be simplified. Taken from https://riptutorial.com/rust/example/4274/read-a-file-as-a-whole-as-a-string.

```rust
let mut content = String::new();
std::fs::File::open(path)
  .and_then(|mut f| f.read_string(&content))
  .map_err(|err| format!("Error reading root CA certs file {:?}: {}", path, err))?;
```